### PR TITLE
Get arducam mipi working

### PR DESCRIPTION
**Pull request by marcojob**
**Prerequisites:**
- OpenCV >4.9.0 compiled with cuda (see intall script in scripts)

**Usage for AR0234 MIPI:**
- Whitebalancing: Can be set using `wb_blue_gain, wb_green_gain, wb_red_gain` in the launch file
- If the gains are set the 0.0, the OpenCV GrayworldWB is used and prints BGR gains. This should be done at the beginning of the experiments
- If the gains are set, the whole pipeline runs on the GPU and only consumes around 35% of one core.

### CODE REVIEW

File: include/usb_cam/formats/bayer.hpp [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+            avg_green_gain = _wb_green_gain;
+            avg_red_gain = _wb_red_gain;
+        }
+
+        // Apply gains using convertTo
+        for (int i = 0; i < 3; ++i) {
+            float gain = (i == 0) ? avg_blue_gain : (i == 1) ? avg_green_gain : avg_red_gain;
+            channels[i].convertTo(channels[i], CV_8U, gain / 256.0);  // 16-bit to 8-bit and apply gain
+        }
+        std::cout << "BGR gains: " << avg_blue_gain << ", " << avg_green_gain << ", "<< avg_red_gain << std::endl;
```
Comment (michaelpantic) [inline include/usb_cam/formats/bayer.hpp]: would remove for now (or add debug topic)

File: include/usb_cam/formats/bayer.hpp [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+    }
+
+    // Compute average gain from deque
+    float _compute_average(const std::deque<float>& gains) {
+        float sum = 0.0f;
+        for (float gain : gains) {
+            sum += gain;
+        }
+        return sum / gains.size();
+    }
```
Comment (michaelpantic) [inline include/usb_cam/formats/bayer.hpp]: wanna store the gains as Eigen::Vector3d?

File: src/usb_cam_node.cpp [hunk 3]
```diff
@@ -210,52 +211,51 @@ class UsbCamNode
...
-      // turn down exposure control (from max of 3)
-      m_camera.set_v4l_parameter("m_exposureauto", 1);
-      // change the exposure level
-      m_camera.set_v4l_parameter("m_exposureabsolute", m_parameters.exposure);
+    // if (m_parameters.auto_white_balance) {
+    //   m_camera.set_v4l_parameter("white_balance_temperature_auto", 1);
+    // } else {
+    //   m_camera.set_v4l_parameter("white_balance_temperature_auto", 0);
+    //   m_camera.set_v4l_parameter("white_balance_temperature", m_parameters.white_balance);
+    // }
```
Comment (michaelpantic) [inline src/usb_cam_node.cpp]: would kick out all commented code

**Review by michaelpantic — APPROVED** [review]
Great work, thanks a lot! See comments, otherwise good to merge.

Could you also rename the repo and cleanout all code currently not used (leave all the format converters though)?

Would mipi_cam be a good name :-)? Or v4l_cam?

**Comment by marcojob** [comment]
> Would mipi_cam be a good name :-)? Or v4l_cam?

Could you give me permission to the repo? I will do the renaming

**Pull request merged**